src/client/components/pages/Courses/CoursesPage.tsx

```diff
@@ -56,10 +61,60 @@ const CoursesPage: FunctionComponent = (): ReactElement => {
     setCourses,
   ] = useState([] as CourseInstanceResponseDTO[]);
 
+  /**
+   * Keeps track of whether a meeting modal is open.
+   * By default, the modal is not visible.
+   */
+  const [
+    meetingModalVisible,
+    setMeetingModalVisible,
+  ] = useState(false);
+
+  /**
+   * Keeps track of the current course instance to be used in the Courses Page
+   * for display and updating purposes
+   */
+  const [
+    currentCourseInstance,
+    setCurrentCourseInstance,
+  ] = useState(null as CoursesPageCourseInstance);
+
   const dispatchMessage = useContext(MessageContext);
 
   const [fetching, setFetching] = useState(false);
 
+  /**
+   * The current ref of the edit meeting button
+   */
+  const meetingEditButtonRef: Ref<HTMLButtonElement> = useRef(null);
+
+  /**
+   * Set the ref focus for the edit meeting button
+   */
+  const setEditButtonFocus = (): void => {
```

Comment:
What do you think about refactoring this to `setFocus(element: Ref<HTMLButtonElement>): void` so you can pass in an arbitrary HTML element and have it focus that?

Reply:
I'm not sure that we would want to do this because for accessibility reasons we want to specifically set the focus back to the edit button that opened the modal initially?